coil-gif/src/main/java/coil/transform/AnimatedTransformation.kt

```diff
@@ -0,0 +1,42 @@
+package coil.transform
+
+import android.graphics.Canvas
+import coil.size.Size
+import android.graphics.PixelFormat as AndroidPixelFormat
+
+/**
+ * An interface for applying transformation on GIFs, animated WebPs, and animated HEIFs.
+ */
+interface AnimatedTransformation {
+
+    /**
+     * Apply transformation on [canvas]
+     *
+     * Note: Do not allocate objects within if you are using [coil.decode.GifDecoder] as this method would be invoked
+     * for each frame.
+     *
+     * @return Opacity of the result after drawing.
+     * [PixelFormat.UNKNOWN] means that the implementation did not change whether the image has alpha. Return
```

Comment:
I don't think we need the pixel format definitions here. Better to point to definitions in the Android SDK.